File: GetPet/GetPet.BusinessLogic/Model/AnimalTraitFilter.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GetPet.BusinessLogic.Model
{
    public class AnimalTraitFilter :  BaseFilter
    {
        public int? AnimalTypeId { get; set; }

        public string AnimalTypeName { get; set; }
    }
}

File: GetPetWebApp/src/app/shared/models/animalTraitFilter.ts
import { BaseFilter } from "./baseFilter";


export class AnimalTraitFilter extends BaseFilter {

    animalTypeId: number = 0;

    animalTypeName: string = '';

    constructor(public page: number, public perPage: number, public createdSince: Date, public id: number) {

        super(page, perPage, createdSince);
        this.animalTypeId = id;
    }
}